File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnDspFeatureGenerator.java

package com.gracenote.gnsdk;

public class GnDspFeatureGenerator {
  private transient long swigCPtr;
  protected transient boolean swigCMemOwn;

  protected GnDspFeatureGenerator(long cPtr, boolean cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnDspFeatureGenerator obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnDspFeatureGenerator(swigCPtr);
      }
      swigCPtr = 0;
    }
  }

/** 
* Initializes the DSP library. 
* @param user set user 
* @param featureType The kind of DSP feature, for example a fingerprint. 
* @param audioSampleRate The source audio sample rate. 
* @param audioSampleSize The source audio sample size. 
* @param audioChannels	The source audio channels 
*/ 
 
  public GnDspFeatureGenerator(GnUser user, GnDspFeatureType featureType, long audioSampleRate, long audioSampleSize, long audioChannels) throws com.gracenote.gnsdk.GnException {
    this(gnsdk_javaJNI.new_GnDspFeatureGenerator(GnUser.getCPtr(user), user, featureType.swigValue(), audioSampleRate, audioSampleSize, audioChannels), true);
  }

/** 
* Use this method to feed audio in to {@link GnDsp} until it returns true 
* @param audioData The source audio 
* @param audioDataBytes The source audio size in bytes 
* @return false : {@link GnDsp} needs more audio, true : {@link GnDsp} received enough audio to generate required feature 
*/ 
 
  public boolean featureAudioWrite(byte[] audioData, long audioDataBytes) throws com.gracenote.gnsdk.GnException {
    return gnsdk_javaJNI.GnDspFeatureGenerator_featureAudioWrite(swigCPtr, this, audioData, audioDataBytes);
  }

/** 
* Indicates the the DSP feature has reached the end of the write operation. 
*/ 
 
  public void featureEndOfAudioWrite() throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnDspFeatureGenerator_featureEndOfAudioWrite(swigCPtr, this);
  }

/** 
* Retrieve {@link GnDspFeature} 
* @return {@link GnDspFeature} 
*/ 
 
  public GnDspFeature featureRetrieve() throws com.gracenote.gnsdk.GnException {
    return new GnDspFeature(gnsdk_javaJNI.GnDspFeatureGenerator_featureRetrieve(swigCPtr, this), true);
  }

}

File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_csharp/src_wrapper/GnDspFeatureGenerator.cs

namespace GracenoteSDK {

public class GnDspFeatureGenerator : global::System.IDisposable {
  private global::System.Runtime.InteropServices.HandleRef swigCPtr;
  protected bool swigCMemOwn;

  internal GnDspFeatureGenerator(global::System.IntPtr cPtr, bool cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = new global::System.Runtime.InteropServices.HandleRef(this, cPtr);
  }

  internal static global::System.Runtime.InteropServices.HandleRef getCPtr(GnDspFeatureGenerator obj) {
    return (obj == null) ? new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnDspFeatureGenerator() {
    Dispose();
  }

  public virtual void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != global::System.IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnDspFeatureGenerator(swigCPtr);
        }
        swigCPtr = new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero);
      }
      global::System.GC.SuppressFinalize(this);
    }
  }

/**
* Initializes the DSP library.
* @param user set user
* @param featureType The kind of DSP feature, for example a fingerprint.
* @param audioSampleRate The source audio sample rate.
* @param audioSampleSize The source audio sample size.
* @param audioChannels	The source audio channels
*/
  public GnDspFeatureGenerator(GnUser user, GnDspFeatureType featureType, uint audioSampleRate, uint audioSampleSize, uint audioChannels) : this(gnsdk_csharp_marshalPINVOKE.new_GnDspFeatureGenerator(GnUser.getCPtr(user), (int)featureType, audioSampleRate, audioSampleSize, audioChannels), true) {
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

  public bool FeatureAudioWrite(byte[] audioData, uint audioDataBytes) {
    bool ret = gnsdk_csharp_marshalPINVOKE.GnDspFeatureGenerator_FeatureAudioWrite(swigCPtr, audioData, audioDataBytes);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
    return ret;
  }

/**
* Indicates the the DSP feature has reached the end of the write operation.
*/
  public void FeatureEndOfAudioWrite() {
    gnsdk_csharp_marshalPINVOKE.GnDspFeatureGenerator_FeatureEndOfAudioWrite(swigCPtr);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

/**
* Retrieve GnDspFeature
* @return GnDspFeature
*/
  public GnDspFeature FeatureRetrieve() {
    GnDspFeature ret = new GnDspFeature(gnsdk_csharp_marshalPINVOKE.GnDspFeatureGenerator_FeatureRetrieve(swigCPtr), true);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
    return ret;
  }

}

}
